src/ImmutableReducerStore.es6.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const each = require('../lib/each.es6');
+const invariant = require('invariant');
+const ImmutableStore = require('./ImmutableStore.es6');
+
+class ImmutableReducerStore extends ImmutableStore {
+
+  constructor(options) {
+    const parentOptions = Object.assign({}, options);
+
+    if (options.init) {
+      parentOptions.init = function init() {
+        options.init.call(this);
+
+        const defaultState = this.defaultState;
+        const keys = Object.keys(this);
+
+        const hasOneImmutableDefaultStateProperty = (
+          ImmutableStore.checkImmutable(defaultState) &&
+          'defaultState' in this &&
+          keys.length === 1
+        );
+
+        invariant(
+          hasOneImmutableDefaultStateProperty,
+          'An ImmutableReducerStore must provide only an Immutable `defaultState` property ' +
+          'during `init`. Instead found the keys `%s`.',
+          keys.join()
+        );
+
+        this.state = this.defaultState;
+        delete this.defaultState;
```

Comment:
This is somewhat opinionated. I'm happy to delete this line if it's considered odd behaviour. For instance, it might be useful to keep `defaultState` around in the cause of a `reset` action being triggered -

```js
        init: function () {
          this.defaultState = ...;
          this.bindActions(Actions.RESET, this.reset);
        },

        private: {
          reset: function (state) {
            return this.defaultState;
          }
        }
```

If our reducers are pure then this seems okay to leave `defaultState` bound to the `ImmutableStoreReducer`. I can simply add an invariant to ensure that the defaultState is never changed once init is called too

Thoughts?